File: src/main/java/com/braintreegateway/TransactionOptionsCloneRequest.java
package com.braintreegateway;

public class TransactionOptionsCloneRequest extends BaseRequest {
    private TransactionCloneRequest parent;
    private Boolean submitForSettlement;

    public TransactionOptionsCloneRequest(TransactionCloneRequest parent) {
        this.parent = parent;
    }

    public TransactionOptionsCloneRequest submitForSettlement(Boolean submitForSettlement) {
        this.submitForSettlement = submitForSettlement;
        return this;
    }

    public TransactionCloneRequest done() {
        return parent;
    }

    @Override
    public String toXmlString() {
        return buildRequest("options").toXmlString();
    }

    @Override
    public String toQueryString() {
        return toQueryString("options");
    }

    @Override
    public String toQueryString(String root) {
        return buildRequest(root).toQueryString();
    }

    protected RequestBuilder buildRequest(String root) {
        return new RequestBuilder(root).addElement("submitForSettlement", submitForSettlement);
    }
}

File: src/main/java/com/braintreegateway/TransactionOptionsCloneRequest.scala
package com.braintreegateway

class TransactionOptionsCloneRequest(val done: TransactionCloneRequest) extends BaseRequest {

  def submitForSettlement(submitForSettlement: Boolean) = {
    _submitForSettlement = submitForSettlement
    this
  }

  override def toXmlString: String = {
    buildRequest("options").toXmlString
  }

  override def toQueryString: String = {
    toQueryString("options")
  }

  override def toQueryString(root: String): String = {
    buildRequest(root).toQueryString
  }

  protected def buildRequest(root: String): RequestBuilder = {
    new RequestBuilder(root).
      addElement("submitForSettlement", _submitForSettlement)
  }

  private var _submitForSettlement: java.lang.Boolean = null
}